zjit/src/hir.rs

```diff
@@ -5299,6 +5299,21 @@ impl ProfileOracle {
     }
 }
 
+fn invalidates_locals(opcode: u32, operands: *const VALUE) -> bool {
+    match opcode {
+        // Control-flow is non-leaf in the interpreter because it can execute arbitrary code on
+        // interrupt. But in the JIT, we side-exit if there is a pending interrupt.
+        YARVINSN_jump
+        | YARVINSN_branchunless
+        | YARVINSN_branchif
+        | YARVINSN_branchnil
+        | YARVINSN_leave => false,
+        // TODO(max): Read the invokebuiltin target from operands and determine if it's leaf
+        YARVINSN_invokebuiltin => true,
```

Comment:
The TODO makes sense, but adding this line now confused me (I needed to check if `rb_zjit_insn_leaf` was wrong on `invokebuiltin`). I would prefer leaving just the TODO comment.

```suggestion
```